soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -122,6 +122,23 @@ void z_mp_entry(void)
 	volatile int ie;
 	uint32_t idc_reg;
 
+	/* Correct the Region Protection Option cachability settings.
+	 * The hardware defaults (everything accessible and uncached)
+	 * don't match the design intent.  Registers in the first
+	 * region are RW but uncached, as is the RAM mapped in the
+	 * fifth.  The same RAM in the sixth is mapped cached.  Note
+	 * that there's actually a HAL call to do this by emulating
+	 * the older cacheattr feature, but it generates significantly
+	 * more code.
+	 */
+	const uint32_t attribs[] = { 2, 15, 15, 15, 2, 4, 15, 15 };
+
+	for (int region = 0; region < 8; region++) {
+		uint32_t addr = 0x20000000 * region;
+
+		__asm__ volatile("wdtlb %0, %1" :: "r"(attribs[region]), "r"(addr));
```

Comment:
It is (was!  finally merged!) an almost three month old PR, that's just going to happen. :)   What happened was that the code just got moved.  It was added in the original spot because it's affirmatively needed for 2.5 (because with no ROM nor architecture assembly, nothing else does it).  Then everything was unified to a single core startup path in a later patch because that's a cleaner design.  You're right that it could have all been done at once, but that would have led to much larger patches and/or a ton of splitting and rework, and I didn't think this was so strange to look at.  It only seems weird when you're looking at it in a single PR.  If you look at the commits in the tree it's a natural evolution.